lib/dal/src/func/authoring.rs

```diff
@@ -0,0 +1,75 @@
+//! This module contains backend functionality for the [`Func`] authoring experience.
+
+#![warn(
+    bad_style,
+    clippy::missing_panics_doc,
+    clippy::panic,
+    clippy::panic_in_result_fn,
+    clippy::unwrap_in_result,
+    clippy::unwrap_used,
+    dead_code,
+    improper_ctypes,
+    missing_debug_implementations,
+    missing_docs,
+    no_mangle_generic_items,
+    non_shorthand_field_patterns,
+    overflowing_literals,
+    path_statements,
+    patterns_in_fns_without_body,
+    unconditional_recursion,
+    unreachable_pub,
+    unused,
+    unused_allocation,
+    unused_comparisons,
+    unused_parens,
+    while_true
+)]
+
+use thiserror::Error;
+
+use crate::attribute::prototype::AttributePrototypeError;
+use crate::func::argument::FuncArgumentError;
+use crate::func::view::FuncViewError;
+use crate::func::FuncKind;
+use crate::{ActionPrototypeError, FuncError, SchemaVariantError};
+
+mod create;
+mod save;
+mod types;
+
+pub use create::create_func;
+pub use create::AttributeOutputLocation;
+pub use create::CreateFuncOptions;
+pub use create::CreatedFunc;
+pub use save::save_func;
+pub use save::SavedFunc;
+pub use types::compile_langjs_types;
+pub use types::compile_return_types;
```

Comment:
API shape is in flux. For now, make everything private and only expose what we need until it becomes stable. I am expecting more structs of the "view" variety in the future or an authoring client. For now, we just use raw funcs.